its/ruling/src/test/java/org/sonar/python/it/PythonExtendedRulingTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2012-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.it;
+
+import com.sonar.orchestrator.Orchestrator;
+import com.sonar.orchestrator.build.SonarScanner;
+import com.sonar.orchestrator.locator.FileLocation;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.python.it.RulingHelper.bugRuleKeys;
+import static org.sonar.python.it.RulingHelper.getOrchestrator;
+
+public class PythonExtendedRulingTest {
+
+  @ClassRule
+  public static final Orchestrator ORCHESTRATOR = getOrchestrator();
+
+  @BeforeClass
+  public static void prepare_quality_profile() throws IOException {
+    List<String> ruleKeys = bugRuleKeys();
+    String pythonProfile = profile("py", "python", ruleKeys);
+    loadProfile(ORCHESTRATOR, pythonProfile);
+  }
+
+  @Test
+  public void test_airflow() throws IOException {
+    executeBuild(buildWithCommonProperties("airflow"));
+  }
+
+  @Test
+  public void test_archery() throws IOException {
+    executeBuild(buildWithCommonProperties("Archery"));
+  }
+
+  @Test
+  public void test_autokeras() throws IOException {
+    executeBuild(buildWithCommonProperties("autokeras"));
+  }
+
+  @Test
+  public void test_black() throws IOException {
+    SonarScanner build = buildWithCommonProperties("black");
+    build.setProperty("sonar.sources", "src");
+    build.setProperty("sonar.tests", "tests");
+    build.setProperty("sonar.test.exclusions", "tests/data/async_as_identifier.py");
```

Comment:
I think we should have this level of configuration for all other projects as well.  We can probably copy it from `peachee-languages`

- This should help us reduce the analysis time, avoiding to run rules on tests files
- I'd expect we have high quality issues from this `extendedRulingTest` , so probably it's better to avoid analyzing test sources and only focus on production code